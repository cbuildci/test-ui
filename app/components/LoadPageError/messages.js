/*
 * LoadPageError Messages
 */

import { defineMessages } from 'react-intl';

export default defineMessages({
    title: {
        id: 'app.components.LoadPageError.title',
        defaultMessage: 'Failed to Load Page',
    },
    header: {
        id: 'app.components.LoadPageError.header',
        defaultMessage: 'Failed to Load Page',
    },
    bodyMessage: {
        id: 'app.components.LoadPageError.bodyMessage',
        defaultMessage: 'An problem was encountered while loading this page.',
    },
});
