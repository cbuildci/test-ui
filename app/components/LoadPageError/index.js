/*
 * LoadPageError
 */

import React from 'react';
import PropTypes from 'prop-types';
import { Helmet } from 'react-helmet';
import { FormattedMessage } from 'react-intl';
import FormattedTitle from 'components/FormattedTitle';
import Panel, { PanelHeader } from 'components/Panel';
import messages from './messages';

export class LoadPageError extends React.PureComponent {
    render() {
        const {
            debugMessage,
        } = this.props;

        // TODO: Use flag to show debug message.

        return (
            <React.Fragment>
                <Helmet>
                    <FormattedTitle {...messages.title}/>
                </Helmet>

                <Panel>
                    <PanelHeader>
                        <FormattedMessage {...messages.header}/>
                    </PanelHeader>

                    <div>
                        <FormattedMessage {...messages.bodyMessage}/>
                    </div>
                </Panel>

                <Panel bandColor="red">
                    {debugMessage != null && (
                        <pre>
                            {debugMessage}
                        </pre>
                    )}
                </Panel>
            </React.Fragment>
        );
    }
}

LoadPageError.propTypes = {
    debugMessage: PropTypes.string,
};

LoadPageError.defaultProps = {
    debugMessage: null,
};

export const pageErrorMessage = (error, info) => {
    let debugMessages = [];

    try {
        if (error) {
            debugMessages.push(`Error Stack\n-------------------------\n${error.stack || error.message || error}`);
        }
    }
    catch (err) {
        // Ignore
    }

    try {
        if (info.componentStack) {
            debugMessages.push(`Component Stack\n-------------------------${info.componentStack}`);
        }
    }
    catch (err) {
        // Ignore
    }

    return (
        <LoadPageError
            debugMessage={debugMessages.length ? debugMessages.join('\n\n') : null}
        />
    );
};

export default LoadPageError;
