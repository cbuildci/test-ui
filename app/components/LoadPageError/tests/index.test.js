import React from 'react';
import { snapshots } from '../../../../internals/testing/snapshot-util';
import LoadPageError from '../index';

describe('<LoadPageError />', () => {
    it('should render expected JSX', () => {
        snapshots(
            <LoadPageError/>
        );
    });

    it('should render expected JSX with debug error message', () => {
        snapshots(
            <LoadPageError
                debugMessage="foo\nbar"
            />
        );
    });
});
