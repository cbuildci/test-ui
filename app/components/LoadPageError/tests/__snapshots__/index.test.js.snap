// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoadPageError /> should render expected JSX 1`] = `
<Fragment>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
  >
    <InjectIntl(FormattedTitle)
      defaultMessage="Failed to Load Page"
      id="app.components.LoadPageError.title"
    />
  </HelmetWrapper>
  <Panel>
    <Panel__PanelHeader>
      <FormattedMessage
        defaultMessage="Failed to Load Page"
        id="app.components.LoadPageError.header"
        values={Object {}}
      />
    </Panel__PanelHeader>
    <div>
      <FormattedMessage
        defaultMessage="An problem was encountered while loading this page."
        id="app.components.LoadPageError.bodyMessage"
        values={Object {}}
      />
    </div>
  </Panel>
  <Panel
    bandColor="red"
  />
</Fragment>
`;

exports[`<LoadPageError /> should render expected JSX 2`] = `
Array [
  <div
    class="Panel-fVAoGK kwqJRB"
  >
    <div
      class="Panel__PanelHeader-cCNZct dLoaXn"
    >
      <span>
        Failed to Load Page
      </span>
    </div>
    <div>
      <span>
        An problem was encountered while loading this page.
      </span>
    </div>
  </div>,
  <div
    class="Panel-fVAoGK KQrbw"
  />,
]
`;

exports[`<LoadPageError /> should render expected JSX with debug error message 1`] = `
<Fragment>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
  >
    <InjectIntl(FormattedTitle)
      defaultMessage="Failed to Load Page"
      id="app.components.LoadPageError.title"
    />
  </HelmetWrapper>
  <Panel>
    <Panel__PanelHeader>
      <FormattedMessage
        defaultMessage="Failed to Load Page"
        id="app.components.LoadPageError.header"
        values={Object {}}
      />
    </Panel__PanelHeader>
    <div>
      <FormattedMessage
        defaultMessage="An problem was encountered while loading this page."
        id="app.components.LoadPageError.bodyMessage"
        values={Object {}}
      />
    </div>
  </Panel>
  <Panel
    bandColor="red"
  >
    <pre>
      foo\\nbar
    </pre>
  </Panel>
</Fragment>
`;

exports[`<LoadPageError /> should render expected JSX with debug error message 2`] = `
Array [
  <div
    class="Panel-fVAoGK kwqJRB"
  >
    <div
      class="Panel__PanelHeader-cCNZct dLoaXn"
    >
      <span>
        Failed to Load Page
      </span>
    </div>
    <div>
      <span>
        An problem was encountered while loading this page.
      </span>
    </div>
  </div>,
  <div
    class="Panel-fVAoGK KQrbw"
  >
    <pre>
      foo\\nbar
    </pre>
  </div>,
]
`;
